wsr-tic-tac-toe/src/app.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from 'react';
+import { st, classes } from './app.st.css';
+import { Button, Modal, MessageBoxFunctionalLayout } from 'wix-style-react';
+import { InfoIcon } from 'wix-style-react';
+import {
+    CREATE_GAME_BOARD,
+    SET_GAME_BOARD_SIZE,
+    BOARD_SIZE,
+    SET_BOARD_SIZE_DESCRIPTION,
+    CONFIRM_TEXT,
+    SET_BOARD_SIZE_INFO_TOOLTIP,
+} from './constants';
+import { InputWithLabel } from 'wix-style-react';
+import {
+    createBoard,
+    generateWinningConditions,
+    removeBoard,
+    setGameState,
+} from './redux/actions';
+import { useDispatch, useSelector } from 'react-redux';
+import { GameBoard } from './components/game-board';
+import { AppState, SquareState, GAME_STATE } from './redux/types';
+
+export interface AppProps {
+    className?: string;
+}
+
+export const App: React.FC<AppProps> = ({ className }) => {
+    const [isModalVisible, setIsModalVisible] = useState<boolean>(false);
+    const [boardSize, setBoardSize] = useState<number>(0);
+    const turnOrder = useSelector((state: AppState) => state.board.turnOrder);
+    const winningScenarios = useSelector((state: AppState) => state.winningScenarios);
+    const boardCreated = useSelector((state: AppState) => state.board.created);
+    const squares = useSelector((state: AppState) => state.board.squares);
+    const dispatch = useDispatch();
+    useEffect(() => {
+        if (boardCreated) checkIfPlayerWon();
+    }, [turnOrder]);
+    function checkIfPlayerWon() {
+        let result: boolean = false;
```

Comment:
I'd really do a separate function/service out of checkIfPlayerWon. And do it only for one player (passed via params).
Especially since you only need to check one player at a time (when turn order changes, you need to check if previous player won. The player, who's turn just came, will not have won yet for sure).